extendeddatacrossword.go

```diff
@@ -259,25 +256,6 @@ func (eds *ExtendedDataSquare) rebuildShares(
 		return nil, false, nil
 	}
 
-	if isExtendedPartIncomplete {
-		// If needed, rebuild the parity shares too.
-		rebuiltExtendedShares, err := eds.codec.Encode(rebuiltShares[0:eds.originalDataWidth])
```

Comment:
[Question] The only potential issue could arise if the parity portion is incomplete and the rebuilt extended shares obtained from `rebuiltShares, err := eds.codec.Decode(shares)` (the second half of  `rebuiltShares`) and `rebuiltExtendedShares, err := eds.codec.Encode(rebuiltShares[0:eds.originalDataWidth])` turn out to be different. Have we conducted any tests to verify this?

In the original code, we initially decode the entire row, and if the extended part (before decoding) is incomplete, we discard that part from the decoded version and replace it with the encoding result. However, in the new version, we retain whatever the decode function provides us with. If we are confident that these two have the same result, then everything looks good to me! (I think they should be the same)

Reply:
> The only potential issue could arise if the parity portion is incomplete

I don't expect this to happen because Decode calls Reconstruct [here](https://github.com/rootulp/rsmt2d/blob/1f1904acc114b41dff838ad1d530feacc1d9f199/leopard.go#L52) and the godoc for Reconstruct claims:

> 	// If there are too few shards to reconstruct the missing
	// ones, ErrTooFewShards will be returned.

Ref: https://github.com/klauspost/reedsolomon/blob/523164698be98f1603cf1235f5a1de17728b2091/reedsolomon.go#L56-L57